Animate AI Assistant button and panel

lukemelia opened the pull request:
https://github.com/cardstack/boxel/assets/353/3b931457-0e38-4a2d-a85f-451868639573

lukemelia committed:
WIP animating AI Assistant panel opening

lukemelia committed:
Add Split View example to boxel-motion test app as a study for the AI Assistant Panel animation

lukemelia committed:
Improve ResizablePanelGroup to support isHidden arg, and for panel to accept splattributes

lukemelia committed:
Animate AI Assistant button and panel

lukemelia committed:
Lint fix

lukemelia committed:
Incoporate boxel-motion builds into CI etc

nickschot commented:
> https://www.loom.com/share/cd5d9bdc3e194ecba4b6a1fb83bc1e6b

I remember that this was still something we have to finalise after finishing the implementation of the scale correction/transform based animating. I don't think the current version has a `fill: forwards` style thing going on by default (if I remember correctly), so it's likely related to `sprite.lockStyles` which locks position and width with inline CSS. Apparently that isn't currently automatically cleaned up after an animation, though we do have an `unlockStyles` function as well. This is called within the `transitionRunner`. You could see if calling it after the animation promise resolves works (for now?). I don't remember if it may have unexpected side effects at this point.

edit: to add to that, I think with the current timeline functionality (animating every sprite for the entire duration instead of just when it's animated), anchoring, padding etc. we may be able to get rid of lockStyles by integrating it into the keyframes.

lukemelia committed:
Generalize `fill` behavior of behaviors rather than special casing StaticBehavior treatment

lukemelia committed:
Default StaticBehavior to a duration of 1, since you can now opt into forward fill

- when using StaticBehavior with SpringBehavior, you don't know how long the animation will be, and you may want a static animation to persist for the duration of the animation

lukemelia committed:
Use context-relative bounds for removed sprites

lukemelia committed:
Simplify split view treatment and adjust animation per Chris

lukemelia commented:
Thanks @nickschot. I updated this PR to use a StaticBehavior to fix the width during the animation. I generalized the fill behavior so that you can opt StaticBehavior into forward filling like the other behaviors (except Wait) already did by default.

Integrating the lockStyles into the timeline is an interesting idea that likely simplifies the overall code and mental model.

lukemelia committed:
Update open state of AI Assistant button

lukemelia commented:
After discussion with the team and with Chris, we decided to defer animation work on host, so I'll close this PR after extracting the non-animation changes from it into new PRs
